Source/WebCore/css/StyleProperties.cpp

```diff
@@ -493,40 +497,28 @@ String StyleProperties::offsetValue() const
     ASSERT(offsetShorthand().length() == 5);
     StringBuilder result;
 
-    auto offsetPosition = propertyAt(findPropertyIndex(CSSPropertyOffsetPosition));
-    if (!offsetPosition.isImplicit())
-        result.append(offsetPosition.value()->cssText());
-
-    auto offsetPath = propertyAt(findPropertyIndex(CSSPropertyOffsetPath));
-    if (!offsetPath.isImplicit())
-        result.append(result.isEmpty() ? "" : " ", offsetPath.value()->cssText());
-
-    // At this point, result is not empty because either offset-position or offset-path
-    // must be present.
-
-    auto offsetDistance = propertyAt(findPropertyIndex(CSSPropertyOffsetDistance));
-    if (!offsetDistance.isImplicit()) {
-        auto offsetDistanceValue = offsetDistance.value();
-        // Only include offset-distance if the distance is non-zero.
-        // isZero() returns std::nullopt if offsetDistanceValue is a calculated value, in which case
-        // we use value_or() to override to false.
-        if (!downcast<CSSPrimitiveValue>(*offsetDistanceValue).isZero().value_or(false))
-            result.append(' ', offsetDistanceValue->cssText());
-    }
-
-    auto offsetRotate = propertyAt(findPropertyIndex(CSSPropertyOffsetRotate));
-    if (!offsetRotate.isImplicit()) {
-        auto offsetRotateValue = offsetRotate.value();
-        if (!downcast<CSSOffsetRotateValue>(*offsetRotateValue).isInitialValue())
-            result.append(' ', offsetRotateValue->cssText());
-    }
+    auto offsetPosition = getPropertyCSSValue(CSSPropertyOffsetPosition);
+    auto offsetPath = getPropertyCSSValue(CSSPropertyOffsetPath);
+    auto offsetDistance = getPropertyCSSValue(CSSPropertyOffsetDistance);
+    auto offsetRotate = getPropertyCSSValue(CSSPropertyOffsetRotate);
+    auto offsetAnchor = getPropertyCSSValue(CSSPropertyOffsetAnchor);
+
+    bool nonInitialDistance = !isInitialValueForLonghand(CSSPropertyOffsetDistance, *offsetDistance);
+    bool nonInitialRotate = !isInitialValueForLonghand(CSSPropertyOffsetRotate, *offsetRotate);
```

Comment:
nit: small personal preference, but I find it easier to follow logic when variables are defined just before their first usage.

Reply:
I tried various organizations of this function, and this was the one I liked best overall.

Reply:
My next patch rewrites this function anyway.